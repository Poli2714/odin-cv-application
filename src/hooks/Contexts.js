import { createContext } from 'react';

const IndexContext = createContext({
  activeIndex: null,
  setActiveIndex: () => {},
});

const KeyContext = createContext({
  activeKey: {
    eduActiveKey: null,
    expActiveKey: null,
  },
  setActiveKey: () => {},
});

const EduErrMsgContext = createContext({
  showEduErrMessage: false,
  setShowEduErrMessage: () => {},
});

const ExpErrMsgContext = createContext({
  showExpErrMessage: false,
  setShowExpErrMessage: () => {},
});

const PersonalInfoContext = createContext({
  personalInfo: {
    profilePicture: '',
    fullName: '',
    currentJobTitle: '',
    email: '',
    mobile: '',
    city: '',
  },
  setPersonalInfo: () => {},
});

const SummaryContext = createContext({
  summary: '',
  setSummary: () => {},
});

const ExperienceContext = createContext({
  experiences: {
    id: {
      jobTitle: '',
      employer: '',
      jobStartDate: '',
      jobEndDate: '',
      jobDescription: '',
    },
  },
  setExperiences: () => {},
});

const EducationContext = createContext({
  educations: {
    id: {
      degree: '',
      university: '',
      eduStartDate: '',
      eduEndDate: '',
      eduInfo: '',
    },
  },
  setEducations: () => {},
});

export {
  EducationContext,
  EduErrMsgContext,
  ExperienceContext,
  ExpErrMsgContext,
  IndexContext,
  KeyContext,
  PersonalInfoContext,
  SummaryContext,
};
