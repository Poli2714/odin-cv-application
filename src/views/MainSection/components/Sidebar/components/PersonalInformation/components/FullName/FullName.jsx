import { useContext } from 'react';

import { InputWrapper } from 'src/components/layout';
import { Label, TextInput } from 'src/components/forms';
import { PersonalInfoContext } from 'src/hooks/Contexts';

function FullName() {
  const { personalInfo, setPersonalInfo } = useContext(PersonalInfoContext);
  const { fullName } = personalInfo;

  const handleChange = e =>
    setPersonalInfo({
      ...personalInfo,
      fullName: e.target.value,
    });

  return (
    <InputWrapper>
      <Label labelName="Full name" labelFor="name" />
      <TextInput
        autofocus={true}
        name="name"
        placeholder="James Owen"
        value={fullName}
        onChange={handleChange}
      />
    </InputWrapper>
  );
}

export default FullName;
