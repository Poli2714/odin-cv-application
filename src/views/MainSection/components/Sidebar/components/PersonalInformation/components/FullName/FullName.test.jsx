import { expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useState } from 'react';

import FullName from './FullName';
import { PersonalInfoContext } from 'src/hooks/Contexts';

vi.mock('src/components/layout', () => ({
  InputWrapper: vi.fn(({ children }) => (
    <div data-testid="input-wrapper">{children}</div>
  )),
}));

vi.mock('src/components/forms', () => ({
  Label: vi.fn(() => <label htmlFor="test">Test label</label>),
  TextInput: vi.fn(({ value, onChange }) => (
    <input
      name="test"
      id="test"
      type="text"
      value={value}
      onChange={onChange}
    />
  )),
}));

function MockParentComponent() {
  const [personalInfo, setPersonalInfo] = useState({ fullName: '' });

  return (
    <PersonalInfoContext.Provider value={{ personalInfo, setPersonalInfo }}>
      <FullName />
    </PersonalInfoContext.Provider>
  );
}

test('renders FullName component', () => {
  render(<FullName />);

  expect(screen.getByTestId('input-wrapper')).toMatchSnapshot();
});

test('updates fullName value as a user types', async () => {
  const user = userEvent.setup();

  render(<MockParentComponent />);
  const input = screen.getByLabelText('Test label');
  expect(input).toHaveValue('');

  await user.type(input, 'Elgun');

  expect(input).toHaveValue('Elgun');
});
