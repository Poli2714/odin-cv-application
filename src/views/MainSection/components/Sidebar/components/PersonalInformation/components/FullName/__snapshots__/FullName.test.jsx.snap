// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders FullName component 1`] = `
<div
  data-testid="input-wrapper"
>
  <label
    for="test"
  >
    Test label
  </label>
  <input
    id="test"
    name="test"
    type="text"
    value=""
  />
</div>
`;
